completions/fish/wl-paste.fish

```diff
@@ -0,0 +1,25 @@
+function __wayland_seats --description 'Print all wayland seats'
+    set -l seat
+
+    if type -q weston-info
+        set seat (weston-info 2>/dev/null | sed -n '/wl_seat/{n;s/\s*name: //;p}')
+    else if type -q loginctl
+        set seat (loginctl --no-legend --no-pager list-seats 2>/dev/null)
+    end
+
+    # Default seat if we have no other commands
+    if  test -z "$seat"
+        set seat "seat0"
+    end
+    
+    echo "$seat"
+end
+
+complete -c wl-paste -s h -l help -d 'Display a help message.'
+complete -c wl-paste -s v -l version -d 'Display version info.'
+complete -c wl-paste -s n -l no-newline -d 'Do not append a newline character.'
+complete -c wl-paste -s l -l list-types -d 'Instead of pasting, list the offered types.'
+complete -c wl-paste -s p -l primary -d 'Use the "primary" clipboard.'
+complete -c wl-paste -s w -l watch -r -d 'Run a command each time the selection changes.'
+complete -c wl-paste -s t -l type -x -d 'Override the inferred MIME type for the content.' -a "(wl-paste -l)"
```

Comment:
Note that bash completions for `wl-paste -t` try to take into account any previous arguments. So if you, say, type `wl-paste -p -s seat1 -t` and try to complete, they'll suggest you types copied into the primary clipboard on that seat. Would it be possible to have the same work with fish completions?

Reply:
Fish's complete I don't believe exposes any context about other arguments passed; We could get all the types for both the primary and selection on all seats delete any duplicates and offer those for an exhaustive list of all types.

Reply:
From glancing at [the fish source](https://github.com/fish-shell/fish-shell/blob/4e931fd967373b0d35676d673c17b4a6d9076f90/src/builtin_complete.cpp#L334-L337), maybe `commandline` would work?